packages/vite/src/node/server/index.ts

```diff
@@ -365,9 +375,25 @@ export async function createServer(
         throw new Error('cannot print server URLs in middleware mode.')
       }
     },
+    async restart(forceOptimize: boolean) {
+      if (server._isRestarting) {
+        return // no multi-restarts
+      }
+      server._isRestarting = true
+      try {
+        if (forceOptimize) {
+          await runOptimize(forceOptimize)
+        }
```

Comment:
Should `runOptimize` be done as part of the `restartServer` process (instead of before)? Since we want to optimize using the newly instantiated plugin instances from the new server.

I'd imagine for `vite-plugin-svelte`'s usecase, whenever the svelte config changes, we restart the vite dev server, to re-instantiate the svelte plugin, to pick up the new svelte config.

Reply:
You are right! I have pushed a fix using an internal flag to pass along the force param to runOptimize.

Also made it so that an existing restart promise is returned to subsequent callers so they can at least await on it being done